### PR TITLE
Add a cancel token to manually enforce a timeout in Quote Requestor

### PR DESCRIPTION
**Pull request by phil-ociraptor**
## Description

Requests made via Axios do not always respect their timeout: https://github.com/axios/axios/issues/647

In order to ensure that all requests adhere to a timeout, we must cancel those requests with a Cancel Token.

## Testing instructions

- Unit test added for this flow

## Types of changes

* Bug fix (non-breaking change which fixes an issue) 

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [x] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 1]
```diff
@@ -254,7 +254,12 @@ export class QuoteRequestor {
                 this._warningLogger(result, 'Invalid RFQ-T indicative quote received, filtering out');
                 return false;
             }
-            if (!hasExpectedAddresses([[makerToken, order.makerToken], [takerToken, order.takerToken]])) {
+            if (
+                !hasExpectedAddresses([
+                    [makerToken, order.makerToken],
+                    [takerToken, order.takerToken],
+                ])
+            ) {
```
Comment (phil-ociraptor) [inline packages/asset-swapper/src/utils/quote_requestor.ts:262]: my prettier did this automatically... I wonder if my settings are different than everyone else's?

File: packages/asset-swapper/src/utils/quote_requestor.ts [hunk 2]
```diff
@@ -392,6 +397,10 @@ export class QuoteRequestor {
 
         const typedMakerUrls = standardUrls.concat(altUrls);
 
+        const cancelTokenSource = axios.CancelToken.source();
```
Comment (alexkroeger) [inline packages/asset-swapper/src/utils/quote_requestor.ts:400]: are there any issues with using the same `cancelToken` across many requests?

Comment (phil-ociraptor) [reply packages/asset-swapper/src/utils/quote_requestor.ts:400]: Excellent question - I was concerned about this myself. 

So I ended up testing this in a toy repo: https://github.com/phil-ociraptor/playground/blob/main/index.js

This functions exactly as if we had a cancel token per request, which was what I did the first time: https://github.com/phil-ociraptor/playground/commit/2aa544dd7870a02996539e3a54eabab1389e1919#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346

File: packages/asset-swapper/test/quote_requestor_test.ts [hunk 3]
```diff
@@ -366,6 +366,93 @@ describe('QuoteRequestor', async () => {
...
+
+            // One request that will timeout
+            mockedRequests.push({
+                ...mockedDefaults,
+                endpoint: 'https://420.0.0.1',
+                responseData: successfulQuote1,
+                callback: async () => {
+                    // tslint:disable-next-line:no-inferred-empty-object-type
+                    return new Promise((resolve, reject) => {
+                        setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (alexkroeger) [reply packages/asset-swapper/test/quote_requestor_test.ts:418]: 👍

File: packages/asset-swapper/src/utils/quote_requestor.ts (hunk 1 above)
Comment (alexkroeger) [reply packages/asset-swapper/src/utils/quote_requestor.ts:262]: it's failing prettier, run `yarn prettier`

**Review by alexkroeger — APPROVED** [review]
Simbot before merging

**Pull request merged**